include/osquery/expected.h

```diff
@@ -0,0 +1,164 @@
+//
+//  expected.hpp
+//  gmock
+//
+//  Created by Max Kareta on 5/15/18.
+//
+
+#pragma once
+
```

Comment:
You are using here some staff from stl, so it would be good to include all of it here explicitly: (`<utility>`, `<type_traits>`, `<memory>`).